From 50e7776cee2b4687381b9077446d540e2a83ba23 Mon Sep 17 00:00:00 2001
From: David Kahurani <redress@duck.com>
Date: Tue, 28 Jan 2025 15:23:47 +0300
Subject: [PATCH] Reduce if/else if statement to a single if statement

This should probably be it, though, it is untested and uncompiled

Signed-off-by: David Kahurani <redress@duck.com>
---
 init/do_mounts.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/init/do_mounts.c b/init/do_mounts.c
index 6af29da88..ee5fef952 100644
--- a/init/do_mounts.c
+++ b/init/do_mounts.c
@@ -513,9 +513,8 @@ struct file_system_type rootfs_fs_type = {
 void __init init_rootfs(void)
 {
 	if (IS_ENABLED(CONFIG_TMPFS)) {
-		if (!saved_root_name[0] && !root_fs_names)
-			is_tmpfs = true;
-		else if (root_fs_names && !!strstr(root_fs_names, "tmpfs"))
+		if ((!saved_root_name[0] && !root_fs_names) ||
+		   (root_fs_names && !!strstr(root_fs_names, "tmpfs")))
 			is_tmpfs = true;
 	}
 }
-- 
2.48.1

